jdaviz/configs/default/plugins/subset_plugin/subset_plugin.py

```diff
@@ -654,32 +699,348 @@ def _set_value_in_subset_definition(self, index, name, desired_key, new_value):
+
+        Parameters
+        ----------
+        regions : list of obj
+            A list of region objects. A region object can be one of the following:
+
+            * Astropy ``regions`` object
+            * ``photutils`` apertures (limited support until ``photutils``
+              fully supports ``regions``)
+            * specutils ``SpectralRegion`` object
+
+        max_num_regions : int or `None`
+            Maximum number of regions to load, starting from top of the list.
+            Default is to load everything.
+
+        refdata_label : str or `None`
+            Label of data to use for sky-to-pixel conversion for a region, or
+            mask creation. Data must already be loaded into Jdaviz.
+            If `None`, defaults to the reference data in the default viewer.
+            Choice of this data is particularly important when sky
+            region is involved.
+
+        return_bad_regions : bool
+            If `True`, return the regions that failed to load (see ``bad_regions``);
+            This is useful for debugging. If `False`, do not return anything (`None`).
+
+        create_new_after : bool
+            If `True`, after subset is created, set subset to be `Create New`
+
+        kwargs : dict
+            Extra keywords to be passed into the region's ``to_mask`` method.
+            **This is ignored if the region can be made interactive.**
+
+        Returns
+        -------
+        bad_regions : list of (obj, str) or `None`
+            If requested (see ``return_bad_regions`` option), return a
+            list of ``(region, reason)`` tuples for region objects that failed to load.
+            If all the regions loaded successfully, this list will be empty.
+            If not requested, return `None`.
+
+        """
+        if len(self.app.data_collection) == 0:
+            raise ValueError('Cannot load regions without data.')
+
+        from photutils.aperture import (CircularAperture, SkyCircularAperture,
+                                        EllipticalAperture, SkyEllipticalAperture,
+                                        RectangularAperture, SkyRectangularAperture,
+                                        CircularAnnulus, SkyCircularAnnulus)
+        from regions import (Regions, CirclePixelRegion, CircleSkyRegion,
+                             EllipsePixelRegion, EllipseSkyRegion,
+                             RectanglePixelRegion, RectangleSkyRegion,
+                             CircleAnnulusPixelRegion, CircleAnnulusSkyRegion)
+        from jdaviz.core.region_translators import regions2roi, aperture2regions
+
+        # If user passes in one region obj instead of list, try to be smart.
+        if (isinstance(regions, SpectralRegion) or
+                (isinstance(regions, list) and isinstance(regions[0], SpectralRegion))):
+            mode = kwargs.pop('mode', None)
+            return self._import_spectral_regions(regions, mode, max_num_regions, return_bad_regions)
```

Comment:
I remember @kecnry was concerned that `apply_roi` is no longer used within the public API to load spectral region, causing deviation between external and internal usage. Indeed, I do not see `_import_spectral_regions` use `apply_roi` at all because `_import_spectral_regions` is only called from image regions. So this concern still stands?

In fact, `create_new_after` is also not used?